### DISCUSSION TITLE
db error: ERROR: tuple concurrently updated

### DISCUSSION THREAD

**Discussion by gorbak25** [electric-sql/electric#559] [Bug reports]
This is an intermittent failure happening when both postgres and sqlite are doing disjoint sets of operations at the same time. Looks like a no-op compensation from sqlite got applied as an UPDATE while postgres was doing something on that row. This error is very rare - like 1 in 20 test runs.

**Comment by kevin-dp** [comment]
Hi @gorbak25, could you please provide more information about the app such that we can try to reproduce the error.
Please also post the relevant client and server logs.

**Comment by balegas** [comment]
@gorbak25 did you figure out the issue or a workaround?

**Comment by gorbak25** [comment]
@balegas Not yet, the total times I observed this problem was 3 times so far. I think it might be related somewhat to my test harness

**Comment by balegas** [comment]
Moving this into a [bug report](https://github.com/electric-sql/electric/discussions/categories/bug-reports) discussion. We will open an issue once we know a way of reproducing it.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
